### PR TITLE
Create the table of tasks component

### PR DESCRIPTION
**Pull request by KhaledToha**

### CODE REVIEW

File: src/interfaces/task.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    id?: number,
+    title: string,
+    description: string,
+    created_at: string,
+    due_date: string,
+    project: string,
+    priority: string,
+    color: string,
+    section: string,
+// eslint-disable-next-line semi
```
Comment (nada-abuzaid) [inline src/interfaces/task.ts:11]: remove this line

File: src/interfaces/task.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export default interface test {
```
Comment (nada-abuzaid) [inline src/interfaces/task.ts:1]: use descriptive name please

File: src/components/tableOfTasks/TaskTable.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+            <TableCell align="right">
+              <Typography variant="subtitle2" color={theme.palette.custom.fontGray}>Due date</Typography>
+            </TableCell>
+
+            <TableCell align="right">
+              <Typography variant="subtitle2" color={theme.palette.custom.fontGray}>Actions</Typography>
+            </TableCell>
+
+          </TableRow>
```
Comment (nada-abuzaid) [inline src/components/tableOfTasks/TaskTable.tsx:84]: what about the following suggestion?

```js
  const KEYS = ['Task name', 'Project', 'Priority', 'Status', 'Due data', 'Actions'];
```

```suggestion
              <TableRow>
              {KEYS.map((value) => (
              <TableCell>
                <Typography variant="subtitle2"  color={theme.palette.custom.fontGray}>
                  {value}
                </Typography>
              </TableCell>
            ))}
          </TableRow>
```

File: src/components/tableOfTasks/TaskTable.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
+import {
+  Typography, Table, TableBody, TableCell, TableContainer, TableHead, TableRow, Box,
+} from '@mui/material';
+import { TaskRow } from '../index.ts';
+import theme from '../../theme/index.ts';
```
Comment (nada-abuzaid) [inline src/components/tableOfTasks/TaskTable.tsx:5]: the `import/extensions` rule is disabled in aslant, so u don't need to type extension.
```suggestion
import { TaskRow } from '../index';
import theme from '../../theme';
```

File: src/components/tableOfTasks/TaskTable.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+      due_date: '2023-07-04T22:00:00.000Z',
+      project: 'PROJECT2',
+      priority: 'low',
+      color: '#06C270',
+      section: 'To-Do',
+    },
+  ];
+  return (
+    <TableContainer component={Box} bgcolor="transparent">
+      <Table sx={{ minWidth: 650 }} size="small" aria-label="a dense table">
```
Comment (nada-abuzaid) [inline src/components/tableOfTasks/TaskTable.tsx:56]: what about the spacing ya Khaled?😆

image


image

File: src/components/tableOfTasks/TaskRow.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        <Typography variant="body1" color={theme.palette.custom.fontGray}>
+          {task.due_date.split('T')[0]}
+        </Typography>
+      </TableCell>
+
+      <TableCell align="right">
+
+        <IconButton>
+          <Box bgcolor="rgba(62, 123, 250, 0.2)" borderRadius={2} padding={0.7}>
+            <MdOutlineEdit style={{ fontSize: 18, color: '#3E7BFA' }} />
```
Comment (nada-abuzaid) [inline src/components/tableOfTasks/TaskRow.tsx:48]: please don't type any color code inside the components style, set it in the theme file
```suggestion
            <MdOutlineEdit style={{ fontSize: 18, color: 'custom.editIcon' }} />
```

**Review by nada-abuzaid — CHANGES_REQUESTED** [review]

File: src/interfaces/task.ts (hunk 1 above)
Comment (KhaledToha) [reply src/interfaces/task.ts:11]: I need it so I can commit , thanks to husky and eslint

File: src/interfaces/task.ts (hunk 2 above)
Comment (KhaledToha) [reply src/interfaces/task.ts:1]: I really like it since you choose it yourself

File: src/components/tableOfTasks/TaskTable.tsx (hunk 3 above)
Comment (KhaledToha) [reply src/components/tableOfTasks/TaskTable.tsx:84]: the thing is that every TableCell has a different style according to our figma style

Comment (nada-abuzaid) [reply src/components/tableOfTasks/TaskTable.tsx:84]: I talk about the table header :)

**Review by nada-abuzaid — APPROVED** [review]

**Review by Amal-Mousa — APPROVED** [review]

**Review by Mohammed-Sbeata — APPROVED** [review]
PRAVOOO

**Review by AyaAljalees — APPROVED** [review]

**Pull request merged**